src/components/MultiStepForm/PlanSelection/PlanSelection.tsx

```diff
@@ -0,0 +1,111 @@
+import { Control, UseFormSetValue, Controller, FieldErrors } from 'react-hook-form';
+import { useState, useEffect } from 'react';
+
+import Switch from '@mui/material/Switch';
+import Typography from '@mui/material/Typography';
+
+import { IFormInput } from '../IFormInput';
+import { monthlyPricingPlans } from './Data/PricingPlansData';
+
+import { SelectPlanButton, PlanSelectionContainer, MonthlyYearlyContainer, OfferParagraph, OfferHeader, OfferExtra } from '../PlanSelection/PlanSelectionStyles';
+import { ErrorsLabel, ErrorContainer } from '../PersonalInformation/PersonalInformationStyles';
+
+interface PlanSelectionProps {
+  setValue: UseFormSetValue<IFormInput>;
+  control: Control<IFormInput>;
+  setYearlyPlan: () => void;
+  isYearlyPlan: boolean;
+  errors: FieldErrors<IFormInput>;
+}
+
+const PlanSelection: React.FC<PlanSelectionProps> = ({ control, setValue, setYearlyPlan, isYearlyPlan, errors }) => {
+  const [activeButtonId, setActiveButtonId] = useState<number | null>(null);
+
+  const yearlyPricingPlans = monthlyPricingPlans.map((plan) => ({
+    ...plan,
+    price: plan.price * 10,
+  }));
+
+  const currentPlans = isYearlyPlan ? yearlyPricingPlans : monthlyPricingPlans;
+
+  useEffect(() => {
+    const buttonId = localStorage.getItem('buttonId');
+    if (buttonId) {
+      setActiveButtonId(JSON.parse(buttonId));
+      const selectedButton: any = currentPlans.find((plan) => plan.id === JSON.parse(buttonId));
+      if (selectedButton) {
+        setValue('selectedPlan', { name: selectedButton.name, price: selectedButton.price });
+      }
+    }
+  }, []);
+
+  const handlePlanSwitch = () => {
```

Comment:
The name of method says "handle plan switch", but it feels like it's resetting the plan?